### PR TITLE
Perezavidien/asp net core masters/session5

### PR DESCRIPTION
**Pull request by perezavidien**
Apologies that I wasn't able to create a PR on time

### CODE REVIEW

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs [hunk 1]
```diff
@@ -11,39 +11,82 @@
...
-            return _itemService.GetAll(userId);
+            var result = _itemService.Get(itemId);
+
+            return Ok(result);
         }
 
         [HttpGet]
-        int Get(int userId)
+        [Route("filterBy")]
+        IActionResult GetByFilters([FromBody] string filters)
```
Comment (jegtugado) [inline ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs:47]: parameter filters should be of type `Dictionary<string, string>`

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs [hunk 2]
```diff
@@ -11,39 +11,82 @@
...
+            _itemService.Add(itemDto);
+
+            return Ok();
+        }
+
+        [HttpPut]
+        [Route("{itemId}")]
+        IActionResult Put(int itemId, [FromBody] ItemUpdateBindingModel itemUpdateModel)
+        {
+            //where to use itemId?
```
Comment (jegtugado) [inline ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs:70]: In restful design it is more appropriate to use the id from the route. The body of the request shouldn't have an id. No need to change just additional info

File: Repositories/DataContext.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using DomainModels;
+using System;
+using System.Collections.Generic;
+
+namespace Repositories
+{
+    public class DataContext
+    {
+        public List<Item> Items { get; set; }
```
Comment (jegtugado) [inline Repositories/DataContext.cs:9]: Please instantiate the Items property because by default it is null and you cannot modify the contents of the list

**Review by jegtugado — CHANGES_REQUESTED** [review]
PR looks good, just some minor changes needed

**Comment by jegtugado** [comment]
I would also like to ask if you were able to submit the session 1 homework which is an essay? This is at the end of the session 1 ppt

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs (hunk 1 above)
Comment (perezavidien) [reply ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs]: Thank you for reviewing  - this was updated.

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs (hunk 2 above)
Comment (perezavidien) [reply ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs]: Thank you for reviewing  - this was updated.

File: Repositories/DataContext.cs (hunk 3 above)
Comment (perezavidien) [reply Repositories/DataContext.cs:9]: Thank you for reviewing  - this was updated.

**Comment by perezavidien** [comment]
Hi @jegtugado - Unfortunately, I wasn't able to submit assignment#1. Can I still submit though?

**Review by jegtugado — APPROVED** [review]

**Comment by jegtugado** [comment]
hi @perezavidien yes you can still submit the homework for session 1

**Pull request merged**